backend/app/main.py

```diff
@@ -364,20 +367,22 @@ async def update_user(userid: str, body: UserPutRequest, current_user: dict = De
             raise HTTPException(status_code=404, detail="User not found")
 
         update_data = body.dict(exclude_unset=True)
+        
+        # checking if email is valid format
+        if "email" in update_data:
+            if not re.match(r"^[a-zA-Z0-9_.+-]+@(utoronto\.ca|mail\.utoronto\.ca)$", update_data["email"]):
+                raise HTTPException(status_code=400, detail="Invalid email format.")
```

Comment:
The openapi spec description needs to be updated to also indicate that this error response is for both invalid id and email.